src/backend/distributed/utils/connection_cache.c

```diff
@@ -201,10 +201,18 @@ bool
 SqlStateMatchesCategory(char *sqlStateString, int category)
 {
 	bool sqlStateMatchesCategory = false;
-	int sqlState = MAKE_SQLSTATE(sqlStateString[0], sqlStateString[1], sqlStateString[2],
-								 sqlStateString[3], sqlStateString[4]);
+	int sqlState = 0;
+	int sqlStateCategory = 0;
+
+	if (sqlStateString == NULL)
```

Comment:
Deserves a comment in the function header, saying that a NULL category is ok, because it's common for certain kind of errors.